src/lwc/utilPlatformEventListener/utilPlatformEventListener.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Created by kenneth.lewis on 2019-06-27.
+ */
+
+import {LightningElement, track, api} from 'lwc';
+import { subscribe, unsubscribe, onError, setDebugFlag, isEmpEnabled } from 'lightning/empApi';
+import { ShowToastEvent } from 'lightning/platformShowToastEvent';
+import { fireEvent } from 'c/pubsubNoPageRef';
+
+export default class PlatformEventListener extends LightningElement {
+    @api channelName = '/event/DeploymentEvent__e'; //Default
+    subscription = {};
+
+    @api isShowToastEnabled = false;
+    @api isConsoleLogEnabled = false;
+    @api isDebugFlagEnabled = false;
+
+    @api _deploymentIds = new Set();
+    _deploymentResponses = new Map();
+
+    connectedCallback() {
+        this.handleSubscribe();
+        setDebugFlag(this.isDebugFlagEnabled);
+    }
+
+    @api
+    registerDeploymentId(deploymentId) {
+        if (this._deploymentResponses.has(deploymentId)) {
+            this.handleEventReceived(this._deploymentResponses.get(deploymentId));
+        } else {
+            this._deploymentIds.add(deploymentId);
+        }
+    }
+
+    showToast(response){
+        const status =
+            response.data.payload.Status__c || response.data.payload.npsp__Status__c;
+        const deploymentId =
+            response.data.payload.DeploymentId__c || response.data.payload.npsp__DeploymentId__c;
+        const evt = new ShowToastEvent({
+            title: 'Deployment completed with Status: ' + status,
+            message: 'Deployment Id: ' + deploymentId,
+            variant: 'success',
+        });
+        this.dispatchEvent(evt);
+    }
+
+    log(response) {
+        const status =
+            response.data.payload.Status__c || response.data.payload.npsp__Status__c;
+        const deploymentId =
+            response.data.payload.DeploymentId__c || response.data.payload.npsp__DeploymentId__c;
+        console.log('Deployment Event received! ' +
+            'Deployment Id: ' + deploymentId +
+            ' with Status: ' + status);
+    }
+
+    isMonitored(deploymentId) {
+        return this._deploymentIds
+            && this._deploymentIds.has(deploymentId);
+    }
+
+    handleSubscribe() {
+        let x = this;
+
+        // Callback invoked whenever a new event message is received
+        const messageCallback = function (response) {
+            console.log('*** ' + 'Received Platform Event' + ' ***');
```

Comment:
```suggestion
```

Reply:
The `log` function above is configurable, but I think we can remove these console.logs from these other functions.